packages/stream-model-instance-handler/src/model-instance-document-handler.ts

```diff
@@ -342,10 +354,47 @@ export class ModelInstanceDocumentHandler implements StreamHandler<ModelInstance
     if (!('immutableFields' in model.content) || model.content.immutableFields.length == 0) return
 
     for (const lockedField of model.content.immutableFields) {
-      const mutated = payload.data.some((entry) => entry.path.split('/').pop() === lockedField)
+      const mutated = payload.data.some(
+        (entry) => entry.path.slice(1).split('/').shift() === lockedField
```

Comment:
this still feels pretty brittle to me, so if we can do @ukstv's suggestion of comparing the before and after content but make sure that we do it in a way that catches type changes, then I think that would be better.  But I'll defer to @ukstv on what he thinks is best.

Reply:
I disagree, it seems much better to me to check the changes payload rather than having custom verification, especially for the cases you described with changes in nested objects and type changes: having custom logic to compare the values might be prone to edge cases or other bugs, while just checking if there is any change operation seems much safer as only keys (which are always strings) need to be compared.

Reply:
Okay fair.  I don't like us having to understand and parse the jsonPatch format, but I do see that that's a more targeted check to see if anything could possibly have failed with the field we care about.  I'm happy either way